code/modules/mob/living/simple_animal/hostile/jungle_mobs.dm

```diff
@@ -578,11 +578,336 @@
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/WarmupAttack()
+	if(combatant_state == SEEDLING_STATE_NEUTRAL)
+		combatant_state = SEEDLING_STATE_WARMUP
+		walk(src,0)
+		update_icons()
+		var/target_dist = get_dist(src,target)
+		var/living_target_check = isliving(target)
+		if(living_target_check)
+			if(target_dist > 7)//Offscreen check
+				SolarBeamStartup(target)
+				return
+			if(get_dist(src,target) >= 4 && prob(40))
+				SolarBeamStartup(target)
+				return
+		addtimer(CALLBACK(src, .proc/Volley), 5)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/SolarBeamStartup(mob/living/living_target)//It's more like requiem than final spark
+	if(combatant_state == SEEDLING_STATE_WARMUP && target)
+		combatant_state = SEEDLING_STATE_ACTIVE
+		living_target.apply_status_effect(/datum/status_effect/seedling_beam_indicator, src)
+		beam_debuff_target = living_target
+		playsound(src,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		if(get_dist(src,living_target) > 7)
+			playsound(living_target,'sound/effects/seedling_chargeup.ogg', 100, 0)
+		solar_beam_identifier = world.time
+		addtimer(CALLBACK(src, .proc/Beamu, living_target, solar_beam_identifier), 35)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Beamu(mob/living/living_target, beam_id = 0)
+	if(combatant_state == SEEDLING_STATE_ACTIVE && living_target && beam_id == solar_beam_identifier)
+		if(living_target.z == z)
+			update_icons()
+			solar_beam = new(src,target,time=5,beam_icon_state="solar_beam",maxdistance = INFINITY,btype=/obj/effect/ebeam/solarbeam,beam_sleep_time = 0.5)
+			INVOKE_ASYNC(solar_beam, /datum/beam.proc/Start)
+			living_target.adjustFireLoss(30)
+			living_target.adjust_fire_stacks(0.2)//Just here for the showmanship
+			living_target.IgniteMob()
+			playsound(living_target,'sound/weapons/sear.ogg', 50, 1)
+			addtimer(CALLBACK(src, .proc/AttackRecovery), 5)
+			return
+	AttackRecovery()
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/Volley()
+	if(combatant_state == SEEDLING_STATE_WARMUP && target)
+		combatant_state = SEEDLING_STATE_ACTIVE
+		update_icons()
+		var/datum/callback/cb = CALLBACK(src, .proc/InaccurateShot)
+		var/i
+		for(i=0, i<13, i++)
+			addtimer(cb, i)
+		addtimer(CALLBACK(src, .proc/AttackRecovery), 14)
+
+/mob/living/simple_animal/hostile/jungle/seedling/proc/InaccurateShot()
+	if(target && !QDELETED(target) && combatant_state == SEEDLING_STATE_ACTIVE)
+		if(get_dist(src,target) <= 3)//If they're close enough just aim straight at them so we don't miss at point blank ranges
+			Shoot(target)
+			return
+		var/turf/our_turf = get_turf(src)
+		var/obj/item/projectile/seedling/readied_shot = new /obj/item/projectile/seedling(our_turf)
+		readied_shot.current = our_turf
+		readied_shot.starting = our_turf
+		readied_shot.firer = src
+		readied_shot.original = target
+		readied_shot.yo = target.y - our_turf.y + rand(-1,1)
+		readied_shot.xo = target.x - our_turf.x + rand(-1,1)
+		readied_shot.fire()
```

Comment:
I don't want actual randomization on the projectile volley. All the shots should be crossing within one tile of the target.